runtime/auth/aws-signing-default/common/src/aws/smithy/kotlin/runtime/auth/awssigning/Canonicalizer.kt

```diff
@@ -193,10 +203,13 @@ private const val STREAM_CHUNK_BYTES = 16384 // 16KB
  */
 internal fun Url.Builder.canonicalPath(config: AwsSigningConfig): String {
     val srcPath = path
-    val srcSegments = srcPath.encodedSegments
-    val destSegments = if (config.useDoubleUriEncode) srcSegments.map(PercentEncoding.Path::encode) else srcSegments
+    val srcSegments = srcPath.segments
+    val mapper: (Encodable) -> Encodable = when (config.useDoubleUriEncode) {
+        true -> { existing -> PercentEncoding.SigV4.encodableFromDecoded(existing.encoded) }
```

Comment:
**correctness**: You're calling `encodableFromDecoded` but passing `existing.encoded`?

Reply:
Ahh right because it's supposed to be double URI encoded?

Reply:
Exactly, this is the double-encoding case. I'll add a comment to make it clearer but this is definitely correct behavior according to all the tests/specs.